### ISSUE TITLE
The parse method is missing.

### ISSUE THREAD

**Issue by nstdio** [mfathi91/persian-date-time#6]
The `PersianDate` class misses the static factory method `parse(CharSequence, DateTimeFormatter)`. It would be nice to have it because of several reasons:

- Eases the `PersianDate` creation from the string.
- Provides familiar API to all `java.time` users and alignment on API level with it's counterpart `LocalDate`.

The `PersianDate::parse(CharSequence, DateTimeFormatter)` arguably parses dates as if it is persian date.
In addition we can have `PersianDate::parseGregorian(CharSequence, DateTimeFormatter)` to parse date as if it is gregorian date and converts it into persian date.

It would be nice also, to have public static factory method `PersianDate::from(TemporalAccessor)`, which will be used at least internally to create `PersianDate` from parsing result.

**Comment by mfathi91** [comment]
Absolutely. Lack of this method was known even while developing this library. But providing such a method requires full support of all of the formats in `DateTimeFormatter`, e.g. `MM yyyy` or `yyyy/MM/dd` and etc for Persian dates. In other words `DateTimeFormatter.ofPattern(...).parse(...)` cannot be used out of the box for Persian Dates.

Having that said, if you have simpler solutions/ideas, I'm open to pull requests.

**Comment by mfathi91** [comment]
Okay, I took another look at the _DateTimeFormatter_ class and realized that I was under the wrong assumption. Well, in order to use _DateTimeFormatter_, one would simply need to pass the correct chronology to it. Something like this:
```java
PersianDate persianDate = DateTimeFormatter.ofPattern("yyyyMMdd").withChronology(PersianChronology.INSTANCE).parse(...);
```
Therefore, lack of `parse` method was due to inadequate knowledge from my side while developing this library.

This method is now added in version 4.1.0.
